packages/xdl/src/detach/Detach.js

```diff
@@ -360,7 +362,7 @@ async function readNullableConfigJsonAsync(projectDir) {
 
 async function prepareDetachedBuildIosAsync(projectDir, args) {
   const config = await readNullableConfigJsonAsync(projectDir);
-  if (config) {
+  if (config && config.exp.name !== SERVICE_CONTEXT_PROJECT_NAME) {
```

Comment:
Do you think it would be worth it to extract the check to a separate function for increased readability and maintainability? Something like `isServiceConfig(exp)` or sth? 🤔

Reply:
I don't think anyone except me will be reading this code as I'll remove it just after release 😛 So I also think there is no need to increase maintainability.